hack/ci/e2e-olm.sh

```diff
@@ -50,30 +63,112 @@ function build_and_push_packages() {
 }
 
-function deploy_olm() {
+function deploy_deps() {
+    # we need OLM
     operator-sdk olm install --version ${OLM_VERSION} --timeout 6m
-}
 
-function deploy_spo() {
     # cert-manager first. This should be done using dependencies in the
     # future
     kubectl apply -f https://github.com/cert-manager/cert-manager/releases/download/v1.10.1/cert-manager.yaml
     kubectl -ncert-manager wait --for condition=ready pod -l app.kubernetes.io/instance=cert-manager
 
-    # let's roll..
+    # All installation methods run off the same catalog
     sed -i "s#gcr.io/k8s-staging-sp-operator/security-profiles-operator-catalog:latest#${CATALOG_IMG}#g" examples/olm/install-resources.yaml
-    kubectl create -f examples/olm/install-resources.yaml
+
+}
+
+function deploy_spo_in_custom_ns() {
+    ns=$1
+    manifests=examples/olm/custom-install-resources.yaml
+
+cat << EOF > $manifests
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: $ns
+  labels:
+    pod-security.kubernetes.io/enforce: privileged
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: CatalogSource
+metadata:
+  name: security-profiles-operator
+  # namespace: openshift-marketplace on OCP
+  namespace: olm
+spec:
+  sourceType: grpc
+  image: $CATALOG_IMG
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: security-profiles-operator-sub
+  namespace: $ns
+spec:
+  channel: stable
+  name: security-profiles-operator
+  source: security-profiles-operator
+  sourceNamespace: olm
+---
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+  name: security-profiles-operator
+  namespace: $ns
+EOF
+
+echo "Installing manifest for custom ns installation.."
+cat $manifests
+
+kubectl create -f $manifests
+}
+
+function deploy_spo() {
+    installation_method=$1
+    manifests=examples/olm/$installation_method-install-resources.yaml
+
+    cp examples/olm/install-resources.yaml $manifests
+
+    case $installation_method in
+    all)
+        ;;
+    own)
+        echo "spec:
+  targetNamespaces:
+  - security-profiles-operator" >> $manifests
+        ;;
+    single)
+        echo "spec:
+  targetNamespaces:
+  - spo-sp-ns" >> $manifests
+        ;;
+    multi)
+        echo "spec:
+  targetNamespaces:
+  - sp-test-1
+  - sp-test-2" >> $manifests
+        ;;
+    esac
+
+    # let's roll..
+    kubectl create -f $manifests
 }
 
 function check_spo_is_running() {
+    ns=$1
+
     # Useful in case the CatalogSource is fubar. We retry several times
     # because on transient errors (which are for some reason common even
     # if the catalog is local) the pod gets restarted
     for i in $(seq 1 5); do
         kubectl -nolm wait --for=condition=ready pods -lolm.catalogSource=security-profiles-operator
-        catalog_logs=$(kubectl -nolm logs $(kubectl -nolm get pods --no-headers -lolm.catalogSource=security-profiles-operator | awk '{print $1}') 2>/dev/null)
-        if [[ -n "$catalog_logs" ]]; then
+        install_rv=$?
```

Comment:
This change is not strictly related to the tests, but I found (because I was running the tests over and over..) that the wait occasionally times out on the first try, but the `catalog_logs` would usually contain something, so I made the retry loop on `install_rv` instead.